src/mono/mono/metadata/class-private-definition.h

```diff
@@ -102,6 +103,8 @@ struct _MonoClass {
 	guint16    *interface_offsets_packed;
 	guint8     *interface_bitmap;
 
+	guint inlinearray_value; /* System.Runtime.CompilerServices.InlineArrayAttribute lenght value */
```

Comment:
Actually... Should we use `gint32`?  The managed `InlineArrayAttribute` type uses `int` for the length.  so it might be negative.  and we will not notice negative values.  (well... it will error out saying they're out of bounds because a negative input will be a very large uint value.  But that's pretty fragile)

Reply:
Right, it should be gint32 to detect negative values as well.